test/browser/lifecycle.js

```diff
@@ -218,6 +218,181 @@ describe('Lifecycle methods', () => {
 		// [should not override state with stale values if prevState is spread within getDerivedStateFromProps](https://github.com/facebook/react/blob/25dda90c1ecb0c662ab06e2c80c1ee31e0ae9d36/packages/react-dom/src/__tests__/ReactComponentLifeCycle-test.js#L1035)
 	});
 
+	describe("#getSnapshotBeforeUpdate", () => {
+		it('should call nested new lifecycle methods in the right order', () => {
```

Comment:
Love this test! This might be out of scope for this PR, but could we move this test to be directly under the "Lifecycle methods" `describe` block or put it in its own `describe` block? This test specificies the behavior of more than just `getSnapshotBeforeUpdate`. It specifies the order of (potentially all) lifecycle methods. I think it would be valuable to expand this test to include the order of the lifecycle methods for all 4 main render "events": mounting, props update, state update, and unmounting.

I started [a test similar to this](https://github.com/andrewiggins/preact/blob/884457b5833799a9f98ad87acbd530430d31eb52/test/browser/lifecycle.js#L59) when implementing `getDerivedStateFromProps` but pulled it out as it was getting messy. This implementation is much more succinct. I'd be happy to expand this test in another PR if you think its useful.

Reply:
That's a great suggestion! Moved the test to the top `describe`-block. I'm a bit hesitant to add `componentWillUpdate`, `componentWillMount` and `componentWillReceiveProps` to the test, because we'll follow react and mark them as deprecated with our next release.

Reply:
Yea, that makes total sense. No need to add them